packaging/greenboot/functions.sh

```diff
@@ -294,3 +294,92 @@ function log_script_exit() {
         echo "FINISHED"
     fi
 }
+
+# lvmsDriverShouldExist shallowly wraps some python yaml parsing to determine if the LVMS-Operator is expected to be
+# deployed.
+# args:
+#   $1 microshift config file path
+# return:
+#   0:  "true", expect the driver to be running
+#   1:  "false", do not expect the driver to be running
+function lvmsDriverShouldExist() {
+    # The python script checks the config's .storage.driver value. Parsing errors are ignored, these will be caught
+    # by MicroShift during startup, before this greenboot script is run. The script favors verbosity over simplified
+    # code.
+    # args:
+    #   None
+    # return:
+    #   None
+    # exit code:
+    #   0: The lvms components should be checked for. This case occurs for any one of:
+    #       - .storage.driver == one of ["", "lvms"]
+    #       - .storage.driver is unset (indicates default config)
+    #       - .storage is unset (indicates default config)
+    #       - the file cannot be opened (indicates default config)
+    #   1: The lvms components should not be checked for. This case occurs for:
+    #       - .storage.driver == "none"
+    if ! python -c \
+    'import yaml, sys
+    try:
+        with open("/etc/microshift/config.yaml") as stream:
+            try:
+                config = yaml.safe_load(stream)
+                if config["storage"]["driver"] == "none":
+                    sys.exit(1)
+                elif config["storage"]["driver"] == "lvms":
+                    sys.exit(0)
+                elif config["storage"]["driver"] == "":
+                    sys.exit(0)
+                else:
+                    sys.exit(1)
+            except KeyError as e:
+                sys.exit(0)
+            finally:
+                stream.close()
+    except Exception as e:
+        print(f"error opening config: {e}", file=sys.Stderr)
+    '; then
+        return 1
+    fi
+}
```

Comment:
Let's add `return 0` in the end explicitly?

Reply:
Since the python has moved to it's own wrapper funcs, I've reverted some of the prior changes that inverted the python exit codes. So the end of them looks like 
```
'; then
    return 0
return 1
```
It's more of a cosmetic change. pointing it out so it so you know why it looks like I implemented the opposite of your suggestion :D